src/app/d2-loadout-builder/process.ts

```diff
@@ -3,13 +3,17 @@ import { sum } from '../util';
 import { D2Item } from '../inventory/item-types';
 import { SetType, LockableBuckets, ArmorSet } from './types';
 import { getSetsForTier } from './utils';
+import { LoadoutBuilder } from './LoadoutBuilder';
 
 let killProcess = false;
 
 /**
  * This safely waits for an existing process to be killed, then begins another.
  */
-export default function startNewProcess(filteredItems: { [bucket: number]: D2Item[] }) {
+export default function startNewProcess(
+  this: LoadoutBuilder,
```

Comment:
oh, that's cool!

Reply:
Yeah, it’s cool! I’m not sure how much I’m a fan of using “this” inside these functions though, rather than just having some callbacks.

Reply:
Yeah definitely placeholder